File: Blog.Data/Record.cs
using Blog.Data.Enums;
using Blog.Data.Identity;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Blog.Data
{
    public class Record
    {
        public int RecordId { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string Content { get; set; }

        public DateTime CreateDate { get; set; }

        [Required]
        public RecordState State { get; set; } = RecordState.Private;

        public int UserId { get; set; }

        public User User { get; set; }

        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    }
}

File: Blog.ReactUI/Src/Types/Records/Record.ts
import RecordState from "Types/Records/RecordState";
import User from "Types/Account/User";
import Comment from "Types/Comments/Comment";

export default class Record {
  public RecordId: number;

  public Name: string;

  public Content: string;

  public CreateDate: Date;

  public State: RecordState;

  public Comments: Comment[];

  public UserId: number;

  public User: User;
}
